mso/resource_mso_schema_template_contract_service_graph.go

```diff
@@ -733,137 +244,25 @@ func resourceTemplateContractServiceGraphUpdate(d *schema.ResourceData, m interf
-
-				prosubnets = append(prosubnets, subnet)
-			}
-			proConnector["subnets"] = prosubnets
-		}
-
-		/*consumerConnector*/
-		conConnector := make(map[string]interface{})
-		conClusterInterface := make(map[string]interface{})
-		conClusterInterface["dn"] = fmt.Sprintf("%s/lIf-%s", dn, node["consumer_connector_cluster_interface"].(string))
-		conConnector["clusterInterface"] = conClusterInterface
-
-		if node["consumer_connector_redirect_policy"] != "" {
-			if node["consumer_connector_redirect_policy_tenant"] == "" {
-				return fmt.Errorf("consumer redirect policy tenant is required")
-			}
-			conRedPolicy := make(map[string]interface{})
-			conRedPolicy["dn"] = fmt.Sprintf("uni/tn-%s/svcCont/svcRedirectPol-%s", node["consumer_connector_redirect_policy_tenant"].(string), node["consumer_connector_redirect_policy"].(string))
-			conConnector["redirectPolicy"] = conRedPolicy
-		}
-
-		if node["consumer_subnet_ips"] != nil {
-			ips := node["consumer_subnet_ips"].([]interface{})
-			consubnets := make([]interface{}, 0, 1)
-			for _, ip := range ips {
-				subnet := make(map[string]interface{})
-				subnet["ip"] = ip.(string)
-
-				consubnets = append(consubnets, subnet)
-			}
-			conConnector["subnets"] = consubnets
-		}
-
-		nodeMap := make(map[string]interface{})
-		nodeMap["serviceNodeRef"] = tempnodeRef
-		nodeMap["providerConnector"] = proConnector
-		nodeMap["consumerConnector"] = conConnector
-
-		siteNodes = append(siteNodes, nodeMap)
-	}
-
-	tempPath := fmt.Sprintf("/templates/%s/contracts/%s/serviceGraphRelationship", TemplateName, contractName)
-	tempConGraph := models.NewTemplateContractServiceGraph("replace", tempPath, serviceGraphRef, templateNodes)
-
-	sitePath := fmt.Sprintf("/sites/%s-%s/contracts/%s/serviceGraphRelationship", siteID, TemplateName, contractName)
-	siteConGraph := models.NewSiteContractServiceGraph("replace", sitePath, serviceGraphRef, siteNodes)
-
-	_, err = msoClient.PatchbyID(fmt.Sprintf("api/v1/schemas/%s", schemaID), tempConGraph, siteConGraph)
-	if err != nil {
-		return err
 	}
-
-	log.Printf("[DEBUG] Completed Update Template Contract Service Graph")
-	return resourceTemplateContractServiceGraphRead(d, m)
+	return templateNodes, nil
 }
 
-func resourceTemplateContractServiceGraphRead(d *schema.ResourceData, m interface{}) error {
-	log.Printf("[DEBUG] Begining Read Template Contract Service Graph")
-	msoClient := m.(*client.Client)
+// setSchemaTemplateContractServiceGraphAttrs sets the resource attributes of the service graph
+// for a template contract in the schema.
+//
+// Parameters:
+// - d: *schema.ResourceData: The resource data containing the schema attributes.
+// - m: interface{}: The client interface.
+// - importFlag: bool: The import flag to determine if the attributes should be imported.
+//
+// Returns:
+// - error: The error encountered, if any.
+func setSchemaTemplateContractServiceGraphAttrs(cont *container.Container, d *schema.ResourceData, importFlag bool) error {
```

Comment:
What is the value of this import_flag? I understand that it is to differentiate between import and read, but should those not do the same thing?

Reply:
The import_flag is used to ignore the resource attributes during the terraform plan.